pkg/controller/replication/controller_utils.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package controller
+package replication
```

Comment:
There are a few things here that might be common to other controllers:
1. Creating/deleting replicas via a pod template 
2. Handling expectations - any controller that modifies a resource and needs to wait on watch notifications for those modifications (so eg: endpoints controller is not in this category, because it creates endpoints but watches pods and services, while the rc and daemon controller are, because they create pods and watch pods).
3. Filter active pods: though this is small enough that it wouldn't be too horrible if other controllers just had a for loop and a bunch of blacklisted states

So I suggest keeping it somewhere common, making it a little more generic if needed (at least to the point where it can support the DC, I believe @AnanyaKumar already has these modifications), and moving just the rc specific functions (updatereplicacount) into the replication module. @mikedanese wdyt?